pkg/anonymization/anonymizer.go

```diff
@@ -0,0 +1,306 @@
+	for _, network := range cidrs {
+		lastIP := network.IP
+		networksInformation = append(networksInformation, subnetInformation{
+			network: *network,
+			lastIP:  lastIP,
+		})
+	}
+
+	return &Anonymizer{
+		configObserver:    configObserver,
+		clusterBaseDomain: strings.TrimSpace(clusterBaseDomain),
+		networks:          networksInformation,
+		translationTable:  make(map[string]string),
+		ipRegex:           regexp.MustCompile(Ipv4Regex),
+	}, nil
+}
+
+// NewAnonymizer creates a new instance of anonymizer with a provided config observer and openshift config client
+func NewAnonymizerFromConfigClient(
+	ctx context.Context, configObserver ConfigProvider, kubeClient kubernetes.Interface, configClient configv1client.ConfigV1Interface,
+) (*Anonymizer, error) {
+	baseDomain, err := utils.GetClusterBaseDomain(ctx, configClient)
+	if err != nil {
+		return nil, err
+	}
+
+	networksConfig, err := configClient.Networks().Get(ctx, "cluster", metav1.GetOptions{})
+	if err != nil {
+		return nil, err
+	}
+
+	var networks []string
+	for _, network := range networksConfig.Spec.ClusterNetwork {
+		networks = append(networks, network.CIDR)
+	}
+	for _, network := range networksConfig.Spec.ServiceNetwork {
+		networks = append(networks, network)
+	}
+	for _, network := range networksConfig.Spec.ExternalIP.AutoAssignCIDRs {
+		networks = append(networks, network)
+	}
+	for _, network := range networksConfig.Spec.ExternalIP.Policy.AllowedCIDRs {
+		networks = append(networks, network)
+	}
+	for _, network := range networksConfig.Spec.ExternalIP.Policy.RejectedCIDRs {
+		networks = append(networks, network)
+	}
+
+	clusterConfigV1, err := kubeClient.CoreV1().ConfigMaps("kube-system").Get(ctx, "cluster-config-v1", metav1.GetOptions{})
+	if err != nil {
+		return nil, err
+	}
+
+	if installConfig, exists := clusterConfigV1.Data["install-config"]; exists {
+		networkRegex := regexp.MustCompile(Ipv4NetworkRegex)
+		for _, network := range networkRegex.FindAllString(installConfig, -1) {
+			networks = append(networks, network)
+		}
+	}
+
+	// we're sorting by subnet lengths, if they are the same, we use subnet itself
+	utils.SortAndRemoveDuplicates(&networks, func(i, j int) bool {
+		if !strings.Contains(networks[i], "/") || !strings.Contains(networks[j], "/") {
+			return networks[i] > networks[j]
+		}
+
+		network1 := strings.Split(networks[i], "/")
+		network2 := strings.Split(networks[j], "/")
+
+		// subnet length
+		if network1[1] > network2[1] {
+			return true
+		} else if network1[1] < network2[1] {
+			return false
+		} else {
+			return network1[0] > network2[0]
+		}
+	})
+
+	return NewAnonymizer(configObserver, baseDomain, networks)
+}
+
+// IsObfuscationEnabled returns true if obfuscation(hiding IP and domain names) is enabled and false otherwise
+func (anonymizer *Anonymizer) IsObfuscationEnabled() bool {
+	if anonymizer.configObserver == nil {
+		return false
+	}
+
+	return anonymizer.configObserver.Config().EnableGlobalObfuscation
```

Comment:
The config will be read only once during operator start right? I think it's OK for now, but we should think about the situation when user enables it and IO is running. Right now it will require IO restart to take some effect.

Reply:
I thought config observer would observe the changes, but it works somehow differently. Considering the comment below we can keep this behaviour and not create the anonymizer at all if the feature is disabled.

Reply:
You need to "listen" on channel from `anonymizer.configObserver.ConfigChanged()` AFAIK.